WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -107,9 +150,46 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
     func didTapEditAd() {
         onEditAd?()
     }
+
+    func loadAISuggestions() async {
+        isLoadingAISuggestions = true
+        errorState = .none
+        tagline = ""
+        description = ""
+
+        do {
+            suggestions = try await fetchAISuggestions()
+            tagline = suggestions.first?.siteName ?? ""
+            description = suggestions.first?.textSnippet ?? ""
+        } catch {
+            DDLogError("⛔️ Error fetching Blaze AI suggestions: \(error)")
+            errorState = .fetchingAISuggestions
+        }
+
+        isLoadingAISuggestions = false
+    }
+
+    func downloadProductImage() async {
+        isDownloadingImage = true
+        if let productImage = await loadProductImage() {
+            image = productImage
+        }
+        isDownloadingImage = false
```

Comment:
Changing the `ProductImage` into `MediaPickerImage` by downloading it. This avoids adding extra logic to deal with both `ProductImage` and `MediaPickerImage` in this view model. 

Downloading the image allows us to deal with one type, `MediaPickerImage`.